.github/workflows/main.yml

```diff
@@ -36,6 +42,9 @@ jobs:
       - name: Checkout our repository
         run: |
           git clone --depth 1 https://github.com/correctcomputation/checkedc-clang ${{github.workspace}}/depsfolder/checkedc-clang
+          cd ${{github.workspace}}/depsfolder/checkedc-clang
+          git fetch --depth 1 origin "${{ github.event.inputs.branch || env.branch_for_scheduled_run }}"
+          git checkout FETCH_HEAD
```

Comment:
```suggestion
          git clone --depth 1 --branch "${{ github.event.inputs.branch || env.branch_for_scheduled_run }}" https://github.com/correctcomputation/checkedc-clang ${{github.workspace}}/depsfolder/checkedc-clang
```

Reply:
I wanted to be able to checkout an arbitrary commit, rather than just branches.

Reply:
Another possibility would be to add a second parameter for the URL of the repository to use.  That might be slightly easier than looking up a commit ID and would display nicely in the job name, and we wouldn't be dependent on GitHub's apparent behavior of making commits from one repository available in all others in the network (which is undocumented AFAIK, though in the special case of a pull request, the target repository has a `refs/N/pull/head` ref pointing to the same commit as the source branch, so that should be OK).  But I don't feel strongly about making this enhancement now.